21-Reinforcement-Learning/interactive.js

```diff
@@ -0,0 +1,52 @@
+class Interactive {
+    constructor(max_step, step, sliderbar, callbacks){
+        this.max_step = max_step;
+        this.step = step;
+        this.callbacks = callbacks; // callbacks respond to update method
+        this.timer = null;
+        this.sliderbar = sliderbar;
+        this.sliderbar.attr('max', this.max_step);
+    }
+    
+    move(){
+        this.sliderbar.property('value', this.step);
+        for(let callback of this.callbacks){
+            callback.update(this.callback_data);
+        }
+    }
+    
+    cleanup(){
```

Comment:
If `cleanup` is only used to cancel the animation, `stop_animation` might make it more clear.